PatternPractic/PatternPractic/ArchiveService.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using PatternPractic.Services;
+
+namespace PatternPractic
+{
+  public class ArchiveService : IArchiveService
+  {
+    public void Archive(object data)

```

Comment:
Может вместо object-ов везде передавать IComponent? Можно будет подробнее залогировать, какой документ обработан
```suggestion
    public void Archive(IComponent data)

```